framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -205,6 +225,32 @@ public void updateForMethodCall(
             MethodInvocationNode n, AnnotatedTypeFactory atypeFactory, V val) {
         ExecutableElement method = n.getTarget().getMethod();
 
+        // List of expressions that this method side-effects (specified as arguments/elements of
+        // @SideEffectsOnly). If the list is empty, then there is no @SideEffectsOnly annotation.
```

Comment:
This choice seems error-prone.  How does this code differentiate between `@SideEffectsOnly({})` and no `@SideEffectsOnly` annotation?  Please add test cases to ensure that they are treated correctly.  That may necessitate changing this code, such as using `null` if there is no `@SideEffectsOnly` annotation.